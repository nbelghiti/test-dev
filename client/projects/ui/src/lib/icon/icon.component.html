<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
