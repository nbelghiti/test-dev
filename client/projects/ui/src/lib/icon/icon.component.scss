// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
