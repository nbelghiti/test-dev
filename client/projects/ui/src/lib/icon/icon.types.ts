export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
