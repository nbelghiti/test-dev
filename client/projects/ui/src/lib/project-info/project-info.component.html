<div class="c-project-info">
	<div *ngIf="imageSrc" class="c-project-info__image-container">
		<img [src]="imageSrc" alt="Project overview model" class="c-project-info__image">
	</div>

	<div class="c-project-info__content">
		<p
			*ngIf="title"
			class="c-project-info__title"
		>
			{{title}}
		</p>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__CREATED_AT' | translate }}</span>

			<span class="c-project-info__value">{{created}}</span>
		</div>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__PHASE' | translate }}</span>

			<ui-status-label
				[type]="phase"
				class="c-project-info__value"
			>
			</ui-status-label>
		</div>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__VAULT' | translate }} #</span>

			<span class="c-project-info__value c-project-info__value--uppercase">{{ vault || '-' }}</span>
		</div>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__CLIENT' | translate }}</span>

			<span class="c-project-info__value">{{ client || '-' }}</span>
		</div>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__LOCATION' | translate }}</span>

			<span class="c-project-info__value">{{ location || '-' }}</span>
		</div>

		<div
			class="c-project-info__item"
		>
			<span class="c-project-info__label">{{ 'PROJECT__FORMAT' | translate }}</span>

			<span class="c-project-info__value">{{ ACCFormat || '-' }}</span>
		</div>
	</div>
</div>
