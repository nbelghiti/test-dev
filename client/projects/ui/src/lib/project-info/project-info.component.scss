// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Project Info
// =============================================================================
.c-project-info {
	border-bottom: $border-width-base solid $border-color-grey-86;
	background-color: $background-color-base;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-project-info__title {
	font-size: $font-size-base;
	line-height: $line-height-base;
	font-weight: $font-weight-bold;

	color: $text-color-base;
}

.c-project-info__content {
	padding: 2rem 3rem;
}

.c-project-info__image-container {
	display: flex;
	justify-content: center;
	align-items: center;

	overflow: hidden;
}

.c-project-info__image {
	max-height: 15rem;
	object-fit: contain;
}

.c-project-info__item {
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 1rem 0;

	border-bottom: $border-width-base solid $border-color-grey-76;

	&:last-child {
		padding-bottom: 0;

		border-bottom: none;
	}
}

.c-project-info__label {
	font-size: $font-size-1x-small;

	color: $text-color-grey-56;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-project-info__value--uppercase {
	text-transform: uppercase;
}
