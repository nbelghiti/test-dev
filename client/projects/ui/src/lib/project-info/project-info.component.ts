import { Component, Input } from '@angular/core';

import { StatusLabelType } from '../status-label/status-label.types';

@Component({
	selector: 'ui-project-info',
	templateUrl: './project-info.component.html',
	styleUrls: ['./project-info.component.scss']
})

export class ProjectInfoComponent {
	@Input() public title: string;
	@Input() public phase: StatusLabelType = StatusLabelType.inProgress;
	@Input() public created: string;
	@Input() public imageSrc: string;
	@Input() public client: string;
	@Input() public location: string;
	@Input() public vault: string;
	@Input() public ACCType: string;
	@Input() public ACCFormat: string;
}
