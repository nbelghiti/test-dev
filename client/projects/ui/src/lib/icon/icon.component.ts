import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
