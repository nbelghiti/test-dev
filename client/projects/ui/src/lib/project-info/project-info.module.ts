import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';

import { IconModule } from '../icon/icon.module';
import { ListModule } from '../list/list.module';
import { StatusLabelModule } from '../status-label/status-label.module';

import { ProjectInfoComponent } from './project-info.component';

@NgModule({
	imports: [
		CommonModule,
		ListModule,
		IconModule,
		TranslateModule,
		StatusLabelModule,
	],
	declarations: [
		ProjectInfoComponent,
	],
	exports: [
		ProjectInfoComponent,
	],
})
export class ProjectInfoModule {}
