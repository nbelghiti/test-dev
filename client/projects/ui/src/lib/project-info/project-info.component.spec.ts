import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';

import { IconComponent } from '../icon/icon.component';
import { StatusLabelComponent } from '../status-label/status-label.component';

import { ProjectInfoComponent } from './project-info.component';

describe('ProjectInfoComponent', () => {
	let component: ProjectInfoComponent;
	let fixture: ComponentFixture<ProjectInfoComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				TranslateModule.forRoot(),
			],
			declarations: [
				IconComponent,
				StatusLabelComponent,
				ProjectInfoComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(ProjectInfoComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
