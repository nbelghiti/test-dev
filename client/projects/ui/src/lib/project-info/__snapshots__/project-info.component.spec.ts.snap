// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfoComponent Should render correctly 1`] = `
<ui-project-info
  phase={[Function String]}
>
  <div
    class="c-project-info"
  >
    
    <div
      class="c-project-info__content"
    >
      
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__CREATED_AT
        </span>
        <span
          class="c-project-info__value"
        >
          
        </span>
      </div>
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__PHASE
        </span>
        <ui-status-label
          class="c-project-info__value"
          ng-reflect-type="in progress"
        >
          <span
            class="c-status-label c-status-label--grey"
            ng-reflect-klass="c-status-label"
            ng-reflect-ng-class="[object Object]"
          >
            <span
              class="c-status-label__icon"
            >
              
            </span>
            <span
              class="c-status-label__icon"
            >
              
            </span>
            <span
              ng-reflect-ng-class="[object Object]"
            >
               in progress 
            </span>
          </span>
        </ui-status-label>
      </div>
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__VAULT #
        </span>
        <span
          class="c-project-info__value c-project-info__value--uppercase"
        >
          -
        </span>
      </div>
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__CLIENT
        </span>
        <span
          class="c-project-info__value"
        >
          -
        </span>
      </div>
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__LOCATION
        </span>
        <span
          class="c-project-info__value"
        >
          -
        </span>
      </div>
      <div
        class="c-project-info__item"
      >
        <span
          class="c-project-info__label"
        >
          PROJECT__FORMAT
        </span>
        <span
          class="c-project-info__value"
        >
          -
        </span>
      </div>
    </div>
  </div>
</ui-project-info>
`;
